### DISCUSSION TITLE
Improve documentation

### DISCUSSION THREAD

**Discussion by svetch** [jeremylvln/Shulker#435] [Ideas]
Hi!

The current documentation for the Shulker Operator, while useful, could be significantly enhanced by including more detailed instructions and examples for several key areas. This would greatly assist users in setting up, managing, and customizing their Minecraft servers using the Shulker Operator. Below are the specific areas where improvements are needed:

## 1. Connecting to the Server's Console:
- Detailed instructions on how users can access the Minecraft server console through the shulker-operator. This should include any prerequisites, step-by-step guidance, and examples of common commands or troubleshooting tips.

## 2. Updating server.properties:
- Guidance on how to update the server.properties file of a Minecraft server managed by shulker-operator.

## 3. Shulker Overrides on itzg Docker Image:
- Comprehensive documentation on what specific configurations or settings Shulker overrides when using the itzg Minecraft server Docker image. This should include details on server.properties and any other configurations affected, providing users with clear insights into the predefined settings and how they can customize them further.

## 4. Documentation About Custom Resource Definitions (CRDs):
- A comprehensive overview of the CRDs used by shulker-operator, including their structure. This would greatly benefit users looking to customize their Minecraft servers beyond the basic setup.

## 5. Shulker-Specific Environment Variables:
-  A detailed list and explanation of all environment variables specific to shulker-operator that can be used to configure and manage the Minecraft server instances. Including default values, possible values, and use cases for each variable would be very helpful.

## 6. Search Functionality in Docs Page:
- Implementing a search feature on the documentation site to allow users to easily find the information they are looking for. This would improve the user experience by making the documentation more accessible and navigable.

I believe that these enhancements will make the shulker-operator even more user-friendly and will lower the barrier to entry for new users.

**Comment by jeremylvln** [comment]
Hi @Svetch!

Thanks a lot for this well explained message. I was thinking the documentation was really fine but you really pointed out things that are clearly missing or misexplained. I'm all in for reworking the documentation. However, I might need some help defining the "table of content" of the documentation, a "raw" list of the points to dig into on each page. Writing is not an issue, but finding the things to write is not the same combat haha.

If you have any suggestion does not hesitate, I will not have the bandwidth to tackle that this week I guess, but I will start thinking of a plan and will keep you up-to-date here :)

In the meantime, do not hesitate to ask questions, I will surely answer them if you can't find the answer yourself.

**Comment by samuelh2005** [comment]
Good news, as of #616 docs now contain search! "6" can be considered done.
